### ISSUE TITLE
TarWriter does not throw an error when adding files to an invalid tar archive

### ISSUE THREAD

**Issue by ayousuf23** [dotnet/runtime#71454]
### Description

When adding files to an invalid tar archive, System.Formats.Tar.TarWriter does not throw an error and replaces the invalid tar archive with a new tar archive.

### Reproduction Steps

Step 1: Create `C:\Users\<user>\Documents\file.tar` as a text file
Step 2: Create `C:\Users\<user>\Documents\myfile.txt`
Step 3: 
```
var archivePath = @"C:\Users\<user>\Documents\file.tar"
System.IO.FileStream archiveStream = new FileStream(archivePath, FileMode.Open, FileAccess.ReadWrite, FileShare.None);
System.Formats.Tar.TarWriter tarWriter = new System.Formats.Tar.TarWriter(archiveStream, System.Formats.Tar.TarFormat.Gnu, true);

var fileToAdd = @"C:\Users\<user>\Documents\myfile.txt"
string entryName = "myfile.txt";
tarWriter.WriteEntry(fileToAdd, entryName);

tarWriter.Dispose();
archiveStream.Dispose();
```

### Expected behavior

`System.Formats.Tar.TarWriter` should throw an error when it is created or when adding the file to the invalid tar archive.

### Actual behavior

`file.tar` is overridden by a new tar archive which contains `myfile.txt`

### Configuration

.NET 7 Preview 5
Windows 11, 10.0.22000

**Comment by ayousuf23** [comment]
@carlossanlop

**Comment by carlossanlop** [comment]
Thanks for opening an issue, @ayousuf23 !

So unlike `ZipArchive`, which is a class that does both reading and writing, `TarWriter` only writes entries to a stream. It does not do any reading of entries in advance, and it will start writing at the `stream.Position` at the moment of invoking `WriteEntry`. Since each entry is independent from all others, and there is no central directory in a Tar archive like in a Zip archive (which needs to be read by `ZipArchive` in advance to know where to write next, or to update or delete existing entries), the `TarWriter` is essentially just a tool for writing tar entries to any stream.

In other words, this behavior is _expected_.

If you want to detect if an archive is valid, you will have to try to iterate it with a `TarReader`. It should throw when calling `GetNextEntry`.

**Comment by ayousuf23** [comment]
@carlossanlop Thank you for the clarification. I'm closing this issue.
